File: NaSt2D-2.0/Tools/f/split.f
c********************************************************************
c     Program for data splitting up in isplit sequences
c     -1. -1. -1. labelled the input end
c--------------------------------------------------------------------
c     Programm zur Datenaufspaltung in isplit Folgen
c     -1. -1. -1. kennzeichnet das Eingabeende
c********************************************************************
      program Split
c
c     isplit = number of splits [2,8]
c
      parameter (n=10000)      
      dimension a(n),b(n),c(n) 
      open (unit=1,file='ein')
      open (unit=2,file='aus')
c
      read(1,*) isplit
      if(isplit.gt.8) stop ' isplit>8 '
      do 10 i=1,n
      read(1,*) a(i),b(i),c(i) 
      if(a(i).eq.-1..and.b(i).eq.-1..and.c(i).eq.-1.) goto 20
   10 continue
   20 in=i-1
      i1=1
      do 30 i=i1,in,isplit
      write(2,*) a(i),b(i),c(i)
   30 continue
      call isp(i1,in,isplit,a,b,c,n)
      if(isplit.eq.2) goto 40
      call isp(i1,in,isplit,a,b,c,n)
      if(isplit.eq.3) goto 40
      call isp(i1,in,isplit,a,b,c,n)
      if(isplit.eq.4) goto 40
      call isp(i1,in,isplit,a,b,c,n)
      if(isplit.eq.5) goto 40
      call isp(i1,in,isplit,a,b,c,n)
      if(isplit.eq.6) goto 40
      call isp(i1,in,isplit,a,b,c,n)
      if(isplit.eq.7) goto 40
      call isp(i1,in,isplit,a,b,c,n)
   40 stop 'ok'
      end
c*********************************************************************
      subroutine isp(i1,in,isplit,a,b,c,n)
c
      dimension a(n),b(n),c(n) 
      write(2,*) ' '
      i1=i1+1
      do 10 i=i1,in,isplit
      write(2,*) a(i),b(i),c(i)
   10 continue
      return
      end
c*********************************************************************

File: NaSt2D-2.0/Tools/c/split.c
/*     split.c */
/********************************************************************/
/*     Program for data splitting up in isplit sequences            */
/*     isplit = number of splits [1,in+1]                           */
/*     -1. -1. -1. labelled the input end                           */
/*------------------------------------------------------------------*/
/*     Programm zur Datenausgabe in isplit Folgen                   */
/*     isplit = Anzahl der Splits [1,in+1]                          */
/*     -1. -1. -1. kennzeichnet das Eingabeende                     */
/********************************************************************/
#include <stdio.h>
#include <stdlib.h>

#define VOL1 1000   /* 1st dimension row    - Zeilenindex  i */
#define VOL2 3      /* 2nd dimension column - Spaltenindex j */

int main(void) 
{
   int    i=0,i1=0,in=0,j,isplit,iprint;
   float  array[VOL1][VOL2];
   FILE   *fpein;
   FILE   *fpaus;
   char   *input="ein";                            /* input file  */
   char   *output="aus";                           /* output file */

   fpein=fopen(input,"r");                         /* opening the files */
   fpaus=fopen(output,"w");
   if ((fpein==NULL)||(fpaus==NULL)) {
      printf("Error on opening the files\n");
      return(-1);                                  /* EXIT_ERROR_input_file */
                                     }

   fscanf(fpein,"%d ", &isplit);                   /* read in isplit */
   for(i=0; i < VOL1; i++) {                       /* read in array */
      for(j=0; j < VOL2; j++) fscanf(fpein,"%f ", &array[i][j]);
      in=i-1;
      if((array[i][0] == array[i][1]) && (array[i][0] == array[i][2])) break;
                           }
   if(isplit < 1 || isplit > in+1) {
      printf("Error isplit out-of-limits\n");
      return(-1);                                  /* EXIT_ERROR_isplit */
                                   }

   for(iprint=1; iprint<=in+1; iprint++) {
      for(i=i1; i <= in; i+=isplit) {              /* write spltted array  */
      for(j=0; j < VOL2; j++) fprintf(fpaus,"%3.8f ",array[i][j]);
      fprintf(fpaus,"\n");
                                    }
   if(isplit == iprint) iprint=in+1;
   i1=i1+1;
   fprintf(fpaus,"      \n");
                                         }

   printf("split EXIT_SUCCESS\n");
   fclose(fpein);
   fclose(fpaus);
   return EXIT_SUCCESS;
}














